lib/CustomFields/utils/useCustomFieldsFetch.js

```diff
@@ -11,8 +11,8 @@ const useCustomFieldsFetch = (okapi, backendModuleId, entityType) => {
   const [customFieldsFetchFailed, setCustomFieldsFetchFailed] = useState(false);
 
   const makeOkapiRequest = useCallback(
-    makeRequest(okapi)(backendModuleId),
-    [backendModuleId, okapi.token, okapi.tenant, okapi.url]
+    (url) => makeRequest(okapi)(backendModuleId)(url),
+    [backendModuleId, okapi]
```

Comment:
This one seems to change the semantics. If the okapi object itself is the same one as before but okapi.token has changed then the callback won't fire, will it?

(We won't see that for now, since the token, tenant and url never do change.)

Reply:
I tried leaving the dependency array alone and changing the call  to
```
(url) => makeRequest({ okapi: { ... } })(backendModuleId)(url),
```
but I had test failures that I could not explain. 

But ... you're right that if the dependency array is only checked with shallow equality, this changes the behavior and won't do. TBH, I didn't think hard about this, but saw the simple `[backendModuleId, okapi]` dependency array elsewhere so I just copied it. I think this behavior is likely OK, but it's worth a closer look to make sure I have not just introduced a sneaky bug. I'll change this PR to "Draft" until I can confirm this is acceptable behavior.

Reply:
Thanks for pushing back on this, @MikeTaylor. I was able to adjust the miragejs config to resolve the test failures, allowing the original dependency array to stay as-is.